### PR TITLE
RSDK-5941 add MIME types to the camera API

### PR DESCRIPTION
**Pull request by bazile-clyde**
# Description
I added an array of strings, each representing the camera's supported frame formats.

# Testing
I tested this manually with a camera module that returns either a `video/h264` or `image/jpeg` frame, depending on which MIME type is requested from the SDK and what's supported in the properties. My module if you're _really_ curious is [here](https://github.com/bazile-clyde/camera-static/blob/7e5e5071f042f87de3587f7e036bb8a337b71295/main.go).

I did something like
```go
	cam, _ := camera.FromRobot(robot, "cam")
	props, _ := cam.Properties(context.Background())

	var ctx context.Context
	if slices.Contains(props.FrameFormats, "video/h264") {
		ctx = gostream.WithMIMETypeHint(context.Background(), "video/h264")
	} else {
		ctx = gostream.WithMIMETypeHint(context.Background(), "image/jpeg")
	}

	stream, _ := cam.Stream(ctx)
	img, _, _ := stream.Next(ctx)
	// do something with img

```

# TODO
- [x] Get scope approved ([RSDK-5965](https://viam.atlassian.net/browse/RSDK-5965))
- [x] Change `frame_formats` to `mime_types` ?

[RSDK-5965]: https://viam.atlassian.net/browse/RSDK-5965?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bazile-clyde** [comment]
@bhaney, @njooma  there's little to review but tagging you both as reviewers as a heads up.

File: proto/viam/component/camera/v1/camera.proto [hunk 1]
```diff
@@ -150,6 +150,8 @@ message GetPropertiesResponse {
   // If camera does not provide distortion parameters, leave the field empty
   // Initializing the parameters with 0-values is considered an error
   DistortionParameters distortion_parameters = 3;
+
+  repeated string mime_types = 4;
```
Comment (njooma) [inline proto/viam/component/camera/v1/camera.proto:154]: Wanna add docstrings? :)

**Review by njooma — APPROVED** [review]
I'll make the appropriate change in the python SDK

File: proto/viam/component/camera/v1/camera.proto (hunk 1 above)
Comment (bazile-clyde) [reply proto/viam/component/camera/v1/camera.proto:154]: Good catch. Done.

**Review by seanavery — APPROVED** [review]
LGTM

**Pull request merged**